### ISSUE TITLE
SearchableGrid row selection not working correctly.

### ISSUE THREAD

**Issue by lcorson-carma** [rlsutton1/VaadinCrud#1]
When the user clicks a mouse button the row is selected, then automatically unselected. Clicking a second time on the same row selects it.
The issue appears to be at https://github.com/rlsutton1/VaadinCrud/blob/f25789cb0846cb2f52e92d44cda106e002d0f8ef/src/main/java/au/com/sutton/vaadin/crud/SearchableGrid.java#L651 To get it working I set the thread to sleep for 200ms before selecting the row again, this was enough to allow the new row to be selected and it not firing other events.
Using the arrow keys to navigate up and down the list works correctly, this issue is only when the user selects a row.

**Comment by bsutton** [comment]
I have seen this issue and interestingly it seems to come and go.

S. Brett Sutton
Noojee Contact Solutions
03 8320 8100

**Comment by lcorson-carma** [comment]
Yes, I do believe it's a timing issue. Adding the Thread sleep has allowed it to work correctly. I'll leave that in place in our code and look back at this some other time (or if it starts to get worse and becomes unusable).
